### DISCUSSION TITLE
Add short breaks within audio output

### DISCUSSION THREAD

**Discussion by wita-sec** [OpenVoiceOS/ovos-buildroot#276] [Q&A] [answered]
I have OVOS speaking a couple of statements to report to me on my smart home status. 
All these statements go right one after the other. Thats a bit uncomprehensive for the user. 
It would be great if some short breaks could be added inbetween all these statements.

**Comment by JarbasAl** [comment]
you can pass wait=True in self.speak method to block there, then you can further add delay with sleep() if needed before the next speak call

otherwise all TTS is queued at once to play in order ASAP

**Comment by wita-sec** [comment]
Thank you for timely response. In thread https://github.com/orgs/OpenVoiceOS/discussions/228 you said, that wait=True enforces a Wait until my own speach end. I gave it a try, but it seams that it is  not working for my purpose. I wanted some break in between a series of audio speech outputs.  
sleep(n)  appears not to be an alternative, because "n" depends on the time used for the previous audio output, and i cannot estimate this. 
Maybe i have missed something?

**Reply by JarbasAl** [reply]
```
self.speak("1")
self.speak("2")
self.speak("3")
```
in this example the synth for "1", "2" and "3" happens ASAP and audio files are queued for playback whenever the previous one finishes  (so they still play in order)

```
self.speak("1", wait=True)
self.speak("2", wait=True)
self.speak("3", wait=True)
```

in this example, "1" is synthesizes and plays, once playback finishes, it starts the synth for "2" and so on. this won't add a delay necessarily, as the only extra time is the synth process which is often cached


```
self.speak("1", wait=True)
sleep(1)
self.speak("2", wait=True)
sleep(1)
self.speak("3", wait=True)
```

this way you can force it to wait 1 second after "1" has finished playback and before moving on to "2"

**Reply by wita-sec** [reply]
Please consider my example below.

```
self.speak(<1st long speech>, wait=True)
# I cannot estimate how long playback will go.
sleep(4)
# sleep 4 secs may not be enough and next playback could be queued 
# when first playback is still underway
self.speak(<2nd long speech>, wait=True)
sleep(20)
# sleep 20 secs might be too long and cause an inappropriate break 
self.speak(<3rd long speech>,  wait=True)
```
What if self.speak() had an  argument that indicates a time to interrupt playback after the previous one in the queue has finished?
e.g. 
```
self.speak(<my long speach>, interrupt=1)     
# interrupt this playback for 1 second. In case it is the first in the queue, interrupt will be ignored.
```

**Reply by JarbasAl** [reply]
You don't need to estimate anything, speak will block until playback ends, your sleep only executes when the full audio has played, you only need to decide how long of a pause you want.  Have you tried it? Is there some bug and it isn't working?

**Reply by wita-sec** [reply]
I have attached a file with some more details about my test, and an extract from audio.log 
To me it appears that self.speak is not blocked until the previous playback ends. 
Maybe I have misunderstood something.

[20240921-discussion276.txt](https://github.com/user-attachments/files/17084489/20240921-discussion276.txt)

**Comment by timonvanhasselt** [comment]
@wita-sec Joining the discussion; what TTS-plugin are you using? A streaming TTS maybe?

**Reply by wita-sec** [reply]
I am not sure how to determine the TTS-plugin type.
Here is what ovos-config tells me:

$ ovos-config get -k tts
Value: {'pulse_duck': False, 'module': 'ovos-tts-plugin-server', 'fallback_module': '', 'ovos-tts-plugin-mimic': {'voice': 
'ap', 'persist_cache': False, 'preloaded_cache': '', 'min_free_percent': 75, 'persist_thresh': 1}, 'ovos-tts-plugin-server': 
{'sentence_tokenize': True, 'voice': 'thorsten-low'}}, found in /tts
Value: {'voice': 'ap', 'persist_cache': False, 'preloaded_cache': '', 'min_free_percent': 75, 'persist_thresh': 1}, found in 
/tts/ovos-tts-plugin-mimic
Value: {'sentence_tokenize': True, 'voice': 'thorsten-low'}, found in /tts/ovos-tts-plugin-server

**Comment by JarbasAl** [comment]
I believe you found a bug! fix coming up https://github.com/OpenVoiceOS/OVOS-workshop/pull/238

**Comment by JarbasAl** [comment, marked as answer]
fixed with ovos-workshop 0.1.3

ovos-bus-client 0.1.1 brings extra logs around this process and also handles older workshop versions

update at least one of those and it should solve the issue.

**Reply by wita-sec** [reply]
Retested with ovos-workshop 0.1.3 and ovos-bus-client 0.1.1:
works fine.

Great, thank you.
